keras/callbacks.py

```diff
@@ -1782,7 +1782,7 @@ def __init__(self,
     elif mode == 'max':
       self.monitor_op = np.greater
     else:
-      if 'acc' in self.monitor:
+      if 'acc' in self.monitor or 'auc' in self.monitor:
```

Comment:
prefer `or self.monitor == 'auc'` to avoid potential collisions

Reply:
The usual value of ```self.monitor``` is ```val_acc``` or ```val_auc``` . So it uses ```in``` here .

Reply:
Then use `endswith`. Otherwise unrelated metrics that have the substring "auc" will get caught.

Reply:
my point was to use `endswith('_acc`)`

Reply:
How about 
```if self.monitor.endswith('acc') or self.monitor.endswith('accuracy') or self.monitor.endswith('auc')```
?

Reply:
Yes, that works

Reply:
done